profile/src/v100/entity/persona/persona_data/persona_data_identified_collection_types.rs

```diff
@@ -0,0 +1,349 @@
+use crate::prelude::*;
+
+macro_rules! declare_collection_of_identified_entry {
+    ($id_ent_type:ty,$struct_name:ident) => {
+        #[derive(
+            Serialize,
+            Deserialize,
+            Clone,
+            PartialEq,
+            Hash,
+            Eq,
+            derive_more::Display,
+            derive_more::Debug,
+            uniffi::Record,
+        )]
+        #[debug("{collection}")]
+        #[display("{}", self.display_string())]
+        #[serde(transparent)]
+        pub struct $struct_name {
+            pub collection: IdentifiedVecVia<$id_ent_type>,
+        }
+
+        impl Default for $struct_name {
+            fn default() -> Self {
+                Self::entries([])
+            }
+        }
+
+        impl std::ops::Deref for $struct_name {
+            type Target = IdentifiedVecVia<$id_ent_type>;
+
+            fn deref(&self) -> &Self::Target {
+                &self.collection
+            }
+        }
+
+        impl $struct_name {
+            pub fn entries<I>(values: I) -> Self
+            where
+                I: IntoIterator<Item = $id_ent_type>,
+            {
+                Self {
+                    collection: IdentifiedVecVia::from_iter(values),
+                }
+            }
+
+            pub fn new(value: $id_ent_type) -> Self {
+                Self::entries([value])
+            }
+
+            /// Creates a new CollectionOf PersonaDataEntries using just the *value*, which will be given a
+            /// generated ID and put in a `$id_ent_type`
+            pub fn single_value(
+                value: <$id_ent_type as PersonaDataEntryValue>::Value,
+            ) -> Self {
+                Self::new(value.into())
+            }
+
+            pub fn values<I>(values: I) -> Self
+            where
+                I: IntoIterator<
+                    Item = <$id_ent_type as PersonaDataEntryValue>::Value,
+                >,
+            {
+                Self::entries(values.into_iter().map(|v| v.into()))
+            }
+        }
+
+        impl $struct_name {
+            fn display_string(&self) -> String {
+                let items =
+                    self.items().into_iter().map(|v| v.to_string()).join(", ");
+                format!("[{}]", items)
+            }
+        }
+
+        impl HasPlaceholder for $struct_name {
+            fn placeholder() -> Self {
+                $struct_name::entries([
+                    <$id_ent_type>::placeholder(),
+                    <$id_ent_type>::placeholder_other(),
+                ])
+            }
+
+            fn placeholder_other() -> Self {
+                $struct_name::new(<$id_ent_type>::placeholder_other())
+            }
+        }
+    };
+}
+
+declare_collection_of_identified_entry!(
```

Comment:
nice usage of macros, took might some minutes to figure out what you did, but this is awesome :)!